libp2p/protocols/pubsub/floodsub.nim

```diff
@@ -51,16 +51,15 @@ method subscribeTopic*(f: FloodSub,
     # unsubscribe the peer from the topic
     f.floodsub[topic].excl(peer)
 
-method handleDisconnect*(f: FloodSub, peer: PubSubPeer) =
+method unsubscribePeer*(f: FloodSub, peer: PubSubPeer) =
   ## handle peer disconnects
   ##
-  
-  procCall PubSub(f).handleDisconnect(peer)
 
-  if not(isNil(peer)) and peer.peerInfo notin f.conns:
-    for t in toSeq(f.floodsub.keys):
-      if t in f.floodsub:
-        f.floodsub[t].excl(peer)
+  procCall PubSub(f).unsubscribePeer(peer)
+
+  for t in toSeq(f.floodsub.keys):
```

Comment:
```suggestion
  for _, v in f.floodsub.mitems(): v.excl(peer)
```

Reply:
This is still prone to table modified issues, even if we don't run into it again, it'll be easy to sneak in later changes.

Reply:
well, reading this code is weird because it looks like modification protection, even though it's not needed - ie I'd expect there to be modifications but there aren't so it's inconsistent

Reply:
Well, also there are no await in the loop, so what @arnetheduck proposed will be fine and faster. But end result equal anyway for now.